packages/mds-stream/stream-kafka.ts

```diff
@@ -0,0 +1,116 @@
+import { VehicleEvent, Telemetry, Device } from '@mds-core/mds-types'
+import log from '@mds-core/mds-logger'
+import Kafka, { ProducerStream } from 'node-rdkafka'
+import { Stream } from './stream-interface'
+
+const {
+  env: { KAFKA_HOST = 'localhost:9092' }
+} = process
+
+/*
+  FIXME:
+  Necessary interface due to already-fixed typedef problem in the node-rdkafka repo,
+  will be removed upon next upstream release
+*/
+interface Producer {
+  createWriteStream(conf: any, topicConf: any, streamOptions: any): ProducerStream
+}
+
+/*
+  FIXME:
+  Strange casting here necessary due to problem described in the Producer interface above
+*/
+const eventStream = ((Kafka.Producer as unknown) as Producer).createWriteStream(
+  {
+    'metadata.broker.list': KAFKA_HOST,
+    'queue.buffering.max.messages': 100000
+  },
+  {},
+  {
+    topic: 'mds.event'
+  }
+)
+
+/*
+  FIXME:
+  Strange casting here necessary due to problem described in the Producer interface above
+*/
+const telemetryStream = ((Kafka.Producer as unknown) as Producer).createWriteStream(
```

Comment:
Seems like a generic wrapper for `createWriteStream` would be useful ... a lot of the options passed in seem not to change (maybe use the partial options pattern).

Reply:
Yep. It's going to be a bit gross looking for now because I don't think the actual typed options are exported by the typedefs for node-rdkafka (hence the gross any-typed interface at the top).